src/ListPool/ValueListPool.cs

```diff
@@ -204,7 +204,35 @@ public void AddRange(ReadOnlySpan<T> items)
         }
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public void AddRange(T[] array) => AddRange(array.AsSpan());
+        public void AddRange(T[] array)
+        {
+            if (_disposableBuffer == null)
+            {
+                AddRange(array.AsSpan());
+                return;
+            }
+
+            int count = Count;
+            T[] buffer = _disposableBuffer;
+
+            foreach (T item in array)
```

Comment:
We should use `Array.Copy` and if `_disposableBuffer` is null we should check if the capacity of the `_buffer` is enough and perform `Span.Copy` otherwise create `_disposableBuffer ` and resize it and use `Array.Copy` there and then update the reference of the `_buffer`.